src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/ImageSource.cs

```diff
@@ -112,7 +112,7 @@ protected static double PixelsToDIPs(double dpi, int pixels)
 
             if (dpif < 0.0F || FloatUtil.IsCloseToDivideByZero(96.0F, dpif))
             {
-                dpif = 96.0F;
+                return pixels;
```

Comment:
Before returning we should typecast pixels to double.

Reply:
@dipeshmsft The int pixels will cast to double automatic

Reply:
Do you mean I should cast to double manual ?

Reply:
I guess we can leave it to that.